# Copyright (c) 2024-2025 Voyant Photonics, Inc.
#
# This example code is licensed under the MIT License.
# See the LICENSE file in the repository root for full license text.

/**:
  ros__parameters:
    ## Live node parameters
    # These are the standard parameters for local testing in loopback mode:
    # - "0.0.0.0:4444": Local binding address and port
    # - "224.0.0.0": Multicast group address
    # - "127.0.0.1": Interface address (localhost/loopback)
    #
    # When connecting to an actual sensor
    #  - "0.0.0.0:4444": Binds to all interfaces on port 4444
    #  - "224.0.0.0": Standard multicast group address for the sensor
    #  - "192.168.20.100": Your network interface's IP address
    binding_address: '0.0.0.0:4444'
    multicast_group: '224.0.0.0'
    interface_address: '127.0.0.1'

    ## Shared parameters
    # Point format selection
    # - 0 = UNKNOWN (will throw error)
    # - 1 = MDL_STANDARD (recommended) - x,y,z,v,snr,drop_reason,timestamp,index
    # - 2 = MDL_EXTENDED - adds calibrated_reflectance, noise_mean_estimate, min_ramp_snr, frame_index
    point_format: 1

    # timestamp_mode[optional]
    # - 0 = TIME_FROM_SENSOR
    # - 1 = TIME_FROM_ROS
    timestamp_mode: 0

    # Enable Single Point Noise filter (+ other filtering of invalid points)
    valid_only_filter: false

    ## The below params are shared with the bin_to_mcap conversion
    frame_id: 'voyant_sensor'
    bin_input: '' # Path to your bin file
    mcap_output: '' # Desired output path
    storage_id: 'mcap' # sqlite3/mcap
    serialization_format: 'cdr'
    topic_name: '/point_cloud' # Changing this will break the foxglove multi colormap visualization,
